### PR TITLE
Feature force reset

### PR DESCRIPTION
**Pull request by mabhi**
### What does this PR change?

- Added feature that navigates user to the password reset screen, if they haven't done so after the identity got created. 

### Does the PR depend on any other PRs or Issues? If yes, please list them.

- [add force reset](https://github.com/paralus/paralus/pull/115)

### Checklist

I confirm, that I have...

- [] Read and followed the contributing guide in `CONTRIBUTING.md`
- [ ] Added tests for this PR
- [x] Formatted the code using `npm run format` (if applicable)
- [ ] Updated [documentation on the Paralus docs site](https://github.com/paralus/website/blob/main/docs) (if applicable)
- [] Updated `CHANGELOG.md`

### CODE REVIEW

**Comment by mabhi** [comment]
@akshay196 please check

**Comment by mabhi** [comment]
- [ ] Complete flow for generic users
- [ ] Navigate users to dashboard instead of login screen after password reset and also provide link to navigate in case user does not want to wait
- [ ] Provide new generated password instead of recovery link and reset `forceReset` flag

**Comment by niravparikh05** [comment]
ensure to run npm run format for every commit

File: src/appMain/routes/users/routes/UsersList/UserList.js [hunk 1]
```diff
@@ -423,10 +426,12 @@ class UserList extends React.Component {
       resetPassword(user.metadata.name)
         .then((resp) => {
           const recoveryLink = resp?.data?.recoveryLink;
+          const recoveryPassword = resp?.data?.defaultPassword;
```
Comment (niravparikh05) [inline src/appMain/routes/users/routes/UsersList/UserList.js:429]: we dont need this

File: src/actions/Users.js [hunk 2]
```diff
@@ -188,6 +190,19 @@ export function editUser(params) {
   };
 }
 
+export function updateForceReset(ud, successcb, errcb) {
+  ud.spec.forceReset = false;
```
Comment (niravparikh05) [inline src/actions/Users.js:194]: we dont need to send user as input

File: src/containers/Auth/KratosSettings.js [hunk 3]
```diff
@@ -35,6 +46,18 @@ const KratosSettings = (props) => {
       })
       .catch(console.error);
 
+  useEffect(() => {
+    props.getUserSessionInfo();
+    if (userid) {
```
Comment (niravparikh05) [inline src/containers/Auth/KratosSettings.js:51]: userid is not required any longer

File: src/containers/Auth/KratosSettings.js [hunk 4]
```diff
@@ -63,27 +86,54 @@ const KratosSettings = (props) => {
...
+            setResetSuccess(true);
+          } else {
+            // this bock executes for all users, when forceReset is true and users are forced to reset their password instead of using the auto generated password
+            let scb = function () {
+              setMessage("Your password has been reset.");
+              setResetSuccess(true);
+            };
+            const { userAndRoleDetail } = props;
+            console.log(userAndRoleDetail);
+            await props.updateForceReset(userAndRoleDetail, scb, handleError);
```
Comment (niravparikh05) [inline src/containers/Auth/KratosSettings.js:109]: we need not sent user details as input

**Review by niravparikh05 — COMMENTED** [review]
few nits

File: src/appMain/routes/users/routes/UsersList/UserList.js (hunk 1 above)
Comment (mabhi) [reply src/appMain/routes/users/routes/UsersList/UserList.js:429]: Removed

File: src/containers/Auth/KratosSettings.js (hunk 3 above)
Comment (mabhi) [reply src/containers/Auth/KratosSettings.js:51]: Removed

File: src/containers/Auth/KratosSettings.js (hunk 4 above)
Comment (mabhi) [reply src/containers/Auth/KratosSettings.js:109]: Removed.

File: src/actions/Users.js (hunk 2 above)
Comment (mabhi) [reply src/actions/Users.js:194]: As user information is already being passed via cookies, removed the above.

**Review by niravparikh05 — APPROVED** [review]
lgtm

**Pull request merged**